src/listener/listener-fabric.js

```diff
@@ -0,0 +1,112 @@
+
+        this.testUtil = testUtil;
+        let args = require(configPath).blockchain;
+        this.testUtil.init(path.join(__dirname, rootPath, args.config));
+        this.peerEventObject = {};
+        this.peerEventObject.eventUrl = listener_config.peerEventUrl;
+        let tlsCert = fs.readFileSync(path.join(__dirname, rootPath, listener_config.peerEventTlscaPath));
+        this.peerEventObject.eventTlsca = tlsCert;
+        this.peerEventObject.eventServerHostName =listener_config.peerEventHostnameOverride;
+        this.peerEventObject.org = listener_config.peerOrg;
+        this.client = new Client();
+        this.client_kafka = client_kafka;
+        this.producer = producer;
+        this.listener_config = listener_config;
+
+    }
+
+    /**
+     * Fetch Blocks from Fabric peer and publish into kafka
+     *
+     */
+    getBlocks() {
+        let self = this;
+        self.client_kafka.on('error', function (error) {
+            log('Kafka client ERROR', error);
+        });
+
+        self.producer.on('ready', function () {
+            Client.newDefaultKeyValueStore({ path: '../hfc/hfc-test-kvs_peerOrg1' }).then((store) => {
+
+                self.client.setStateStore(store);
+                return self.testUtil.getSubmitter(self.client, true, self.peerEventObject.org);
+
+            }).then((admin) => {
+
+                self.client._userContext = admin;
+                let eh = self.client.newEventHub();
+                eh.setPeerAddr(
+                    self.peerEventObject.eventUrl,
+                    {
+                        pem: Buffer.from(self.peerEventObject.eventTlsca).toString(),
+                        'ssl-target-name-override': self.peerEventObject.eventServerHostName,
+                        'request-timeout': 12000000,
+                        'grpc.max_receive_message_length': -1
+                    }
+                );
+                eh.connect();
+
+                eh.registerBlockEvent((block) => {
+                    let event_data = {};
+                    event_data.validTime = Date.now();
+                    event_data.block = block;
+                    let payload = [{
+                        topic: self.listener_config.topic,
+                        messages: JSON.stringify(event_data),
+                        partition: 0,
+                        attributes: 1
+                    }];
+
+                    self.producer.send(payload, function (error, result) {
+                        if (error) {
+                            log('Error while publishing block in kafka', error);
+                        }
+                    });
+
+                },
+                (err) => {
+                    log('Error in chaincode Event listener :', err);
+                }
+                );
+            });
+
+        });
+        self.producer.on('error', function (error) {
+            log('Producer is not ready', error);
```

Comment:
log should be replaced with Util.getLogger();  when logging errors with logger=Util.getLogger(), please use logger.error().